api/chat.go

```diff
@@ -68,6 +76,106 @@ func configGinChatRouter(router *gin.RouterGroup) {
 	})
 }
 
+func handleSubmitPollOptionVote(ctx tools.TUMLiveContext, msg []byte) {
+	var req submitPollOptionVote
+	if err := json.Unmarshal(msg, &req); err != nil {
+		log.WithError(err).Warn("could not unmarshal submit poll answer request")
+		return
+	}
+	if ctx.User == nil {
+		return
+	}
+
+	if err := dao.AddChatPollOptionVote(req.PollOptionId, ctx.User.ID); err != nil {
+		return
+	}
+
+	voteCount, _ := dao.GetPollOptionVoteCount(req.PollOptionId)
+
+	voteUpdateMap := gin.H{
+		"pollOptionId": req.PollOptionId,
+		"votes":        voteCount,
+	}
+
+	if voteUpdateJson, err := json.Marshal(voteUpdateMap); err == nil {
+		broadcastStream(ctx.Stream.ID, voteUpdateJson)
```

Comment:
Yes you are right! I currently throw them away in front end for normal users which is actually not secure (if the data is confidential). I thought as the data will be published anyways, that would be the easiest approach, but it would be much better to only send it to admins, I will take a look how to do that :)

Reply:
Okay well, that was easy, just replaced by `broadcastStreamToAdmins` 👍